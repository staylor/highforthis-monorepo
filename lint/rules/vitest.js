import { OFF, ERROR } from './constants.js';

export default {
  'vitest/consistent-test-it': [
    ERROR,
    {
      fn: 'it',
      withinDescribe: 'it',
    },
  ],
  'vitest/max-expects': OFF,
  'vitest/no-conditional-expect': OFF,
  'vitest/no-conditional-in-test': OFF,
  'vitest/no-disabled-tests': OFF,
  'vitest/no-hooks': OFF,
  'vitest/prefer-called-with': OFF,
  'vitest/prefer-expect-assertions': OFF,
  'vitest/prefer-strict-equal': OFF,
  'vitest/require-hook': OFF,
  'vitest/prefer-to-be-falsy': OFF,
};
