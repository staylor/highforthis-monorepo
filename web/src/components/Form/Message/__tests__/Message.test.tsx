import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { describe, it, expect } from 'vitest';

import Message from '@/components/Form/Message';

describe('message', () => {
  it('empty', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/foo?message=updated']}>
        <Message />
      </MemoryRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('text', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/foo?message=updated']}>
        <Message text="This is an admin message." />
      </MemoryRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
