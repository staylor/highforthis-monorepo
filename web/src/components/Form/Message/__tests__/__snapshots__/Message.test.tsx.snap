// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`message > empty 1`] = `null`;

exports[`message > text 1`] = `
<div
  class="border-pink border-l-4 bg-neutral-50 shadow-sm relative mb-4 mt-1 block py-px pl-3 pr-9"
>
  <p
    class="text-dark my-2 p-0.5 text-sm"
  >
    This is an admin message.
  </p>
  <button
    class="m-0 cursor-pointer border-none bg-none p-2 absolute right-1 top-0 dashicons-before before:content-dismiss before:text-base before:block before:h-5 before:w-5 before:bg-none"
    type="button"
  />
</div>
`;
